### DISCUSSION TITLE
Custom type problem

### DISCUSSION THREAD

**Discussion by imcsk8** [diesel-rs/diesel#3125] [Q&A]
Hello I have a problem creating a custom Xml Postgres type:
This is the code of my custom type:
```
use diesel::*;
use diesel::deserialize::{self, FromSql};
use diesel::pg::Pg;
use diesel::serialize::{self, IsNull, Output, ToSql};
use diesel::query_builder::QueryId;
use std::io::Write;


// PostgreSQL XML type
#[derive(Debug, Clone, Copy, Default, QueryId, SqlType)]
#[diesel(postgres_type(oid = 142, array_oid = 143))]
pub struct Xml;

#[cfg(all(feature = "postgres_backend"))]
impl ToSql<Xml, Pg> for Xml {
    fn to_sql<W: Write>(&self, out: &mut Output<W, Pg>) -> serialize::Result {
        out.write_all()
    }
}

#[cfg(all(feature = "postgres_backend"))]
impl FromSql<Xml, Pg> for Xml {
    fn from_sql(bytes: Option<&[u8]>) -> deserialize::Result<Self> {
        Ok(bytes)
    }
}

```

But when I compile the project I get this error:

```
error: cannot find attribute `diesel` in this scope
  --> proj/src/my_types.rs:11:3
   |
11 | #[diesel(postgres_type(oid = 142, array_oid = 143))]

```

Do someone has a hint on how to solve this?
Thanks!!

**Comment by weiznich** [comment]
The error message and your code seems to suggest that you use diesel 1.4.8. The corresponding `#[diesel(postgres_type(…)]` attribute is only available on the unreleased master branch. The 1.4.8 release supports a plain `#[postgres(…)]` attribute in this place.
Checkout the corresponding example on the [1.4.x branch](https://github.com/diesel-rs/diesel/blob/1.4.x/diesel_tests/tests/custom_types.rs#L19) for a complete example.

For future questions: Please always add details about which diesel version you are using, as it's otherwise hard to answer questions.
